src/rpc/rawtransaction.cpp

```diff
@@ -814,42 +815,148 @@ UniValue consolidateunspent(const UniValue& params, bool fHelp)
+ * <sr><pl><pk><pl><pk><pl><pk>...<ts><cm>
+ *
+ * sr: Signatures required in OP code
+ * pl: Pubkey length
+ * pk: Pubkey
+ * ts: Total signable pubkeys in OP code
+ * cm: OP code OP_CHECKMULTISIG
+ *
+ * The wallet/user will provide us with the redeemscript so we will know the size of the script in bytes and we add 4 bytes padding in the tx.
+ *
+ * Standard transaction begin with:
+ *   02    000000   ########    01
+ * <txver><padding><MISCDATA><#ofvins>
+ *
+ * The start of a transaction contains the tx version of the transaction followed by 3 bytes of OP_0 which is padding.
+ * MISCDATA then is placed (likely the transaction time etc) followed by the amount of inputs in the transaction.
+ * We can count on the size to be unchanged since no other data will be placed in that area.
+ *
+ * The start of the transaction will be 9 bytes.
+ *
+ * Vin data in a transaction is stored as the following:
+ * (....)  02    000000     fc00   (.......)(................)ffffffff
+ * <txid><vout#><padding><MISCDATA><sigdata><redeemscriptdata><padding>
+ *
+ * txid: the txid on the input being used
+ * vout#: the vout on the txid
+ * padding: 3 bytes of 0x00 at beginning of vin and 4 bytes of 0xff to signify sequence at end of each vin (
+ * MISCDATA: Unknown but consitant across the board.
+ * sigdata: Signature data from canonical signature as mentioned above.
+ * redeemscript: redeemscript of the multisignature address so the wallet/network knows which addresses are involved in multisignature address.
+ *
+ * From this we can gather that the vin size will be 32 + 1 + 3 + 2 + BSS + RSS + 4.
+ * Or simplified to total vin size of all vins: TVIS = (38 + BSS + RSS + 4) * number_inputs
+ *
+ * Vout data in transaction is simply put since it is returning to the same address.
+ *     1     (...10...) 000000   17  a9  14 (......) 87 0000000000
+ * <#ofvouts><MISCDATA><padding><ss><op><ds><script><op><padding>
+ *
+ * MISCDATA: From testing; This can be bigger but only when multiple outputs are present. Here it is 5 bytes.
+ * padding: Start is 3 bytes of 0x00 / end is 5 bytes of 0x00
+ * ss: Size of the script with op codes. 17 is 23 bytes
+ * op: OP code a9 meaning OP_HASH160 / Op code 87 means OP_EQUAL
+ * ds: Size of script. 14 is 20 bytes
+ * script: Script for the output
+ *
+ * From this we can gather that the vout size will be TVOUTS = 1 + 5 + 3 + 23 + 5
+ * Or simplified TVOUTS = 37 bytes
+ *
+ * We can calculate the max amount of inputs that can be used in a consolidation transaction.
+ * We have to respect that the gui has a limit of 32767 max characters. To be safe set max hex size of 30000 or 15000 bytes.
+ *
+ * Formula:
+ * 15000 = ((38 + BSS + RSS + 4) * X) + 37
+ *
+ * Example of a 2 of 3 multisig.
+ * RSS = 105 bytes
+ * BSS = 150 bytes
+ *
+ * 15000 = ((38 + 150 + 105 + 4) * X) + 37
+ * 15000 = (297 * X) + 37
+ * 15000 = 297X + 37
+ * 15000 - 37 = 297X + 37 - 37
+ * 14963 = 297X
+ * -----   ----
+ * 297     297
+ * 50.505050505 = X
```

Comment:
Actually 50.38047... but you should still round up.

Reply:
forgot to change that when i modified formula lol